### PR TITLE
Fix resource_config_path for generic_oauth

### PR DESCRIPTION
**Pull request by sorcix**
# Description

The code in `resource_default_config_path` replaces all underscores with
dots, while Grafana expects expects to find [`auth.generic_oauth`](https://grafana.com/docs/grafana/latest/auth/generic-oauth/).

## Issues Resolved

List any existing issues this PR resolves

## Check List

- [x] A summary of changes made is included in the CHANGELOG under `## Unreleased`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable.

### CODE REVIEW

**Comment by damacus** [comment]
@sorcix thanks for the contribution, let me just fix up the CI. It looks like we're not using the new reusable workflows yet! 

I'l need to merge in #410 first and then we can rebase on that 👍🏼

File: resources/config_auth_generic_oauth.rb [hunk 1]
```diff
@@ -67,3 +67,7 @@
     accumulator_config(:set, rp.to_s.delete_prefix('oauth_'), new_resource.send(rp))
   end
 end
+
+def resource_config_path_override
+  %w(auth.generic_oauth)
```
Comment (damacus) [inline resources/config_auth_generic_oauth.rb:72]: Why are we using an array here rather than just a string?

Comment (bmhughes) [reply resources/config_auth_generic_oauth.rb:72]: This is correct as the code expects an Array so it can deal with nested paths.

**Review by bmhughes — APPROVED** [review]

**Comment by kitchen-porter** [comment]
Released as: [10.0.3](https://github.com/sous-chefs/grafana/releases/tag/10.0.3)

**Comment by bmhughes** [comment]
Thanks for the fix @sorcix, merged!

**Pull request merged**